src/parquet/file/metadata.h

```diff
@@ -178,6 +178,8 @@ class PARQUET_EXPORT FileMetaData {
   // Return const-pointer to make it clear that this object is not to be copied
   const SchemaDescriptor* schema() const;
 
+  std::unordered_map<std::string, std::string> key_value_metadata() const;
```

Comment:
Since Arrow is a build dependency, we could have a common implementation in arrow/util. If that isn't too fancy